Source/Android/arcana/threading/task_schedulers.h

```diff
@@ -20,4 +27,95 @@ namespace arcana
             }
         } threadpool_scheduler{};
     }
+
+    class looper_scheduler final
+    {
+    public:
+        looper_scheduler(looper_scheduler&) = delete;
+
+        explicit looper_scheduler(ALooper* looper)
+            : m_looper{ looper }
+        {
+            if (pipe(m_fd) == -1)
+            {
+                throw std::runtime_error{ std::strerror(errno) };
+            }
+
+            ALooper_acquire(m_looper);
+
+            if (ALooper_addFd(looper, m_fd[0], ALOOPER_POLL_CALLBACK, ALOOPER_EVENT_INPUT, looper_callback, nullptr) == -1)
+            {
+                throw std::runtime_error{ "Failed to add file descriptor to looper" };
+            }
+        }
+
+        ~looper_scheduler()
+        {
+            if (m_looper != nullptr)
```

Comment:
How would this happen?

Reply:
Move constructor can make the old version nullptr.